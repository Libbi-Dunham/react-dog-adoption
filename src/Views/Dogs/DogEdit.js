import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import DogForm from '../../Components/DogCard/DogForm';
import { getDogsById, updateDog } from '../../services/DogRoute';
import { useHistory } from 'react-router-dom';

export default function DogEdit() {
  const [name, setName] = useState('');
  const [bio, setBio] = useState('');
  const [image, setImage] = useState('');
  const [age, setAge] = useState('');
  const [breed, setBreed] = useState('');
  const history = useHistory();

  const params = useParams();
  useEffect(() => {
    const fetchData = async () => {
      const response = await getDogsById(params.id);
      setName(response.name);
      setBio(response.bio);
      setImage(response.image);
      setAge(response.age);
      setBreed(response.breed);
    };
    fetchData();
  }, [params.id]);

  const handleSubmit = async (e) => {
    try {
      e.preventDefault();
      await updateDog(params.id, name, bio, image, age, breed);
      alert('You Have Successfully updated A dog!');
    } catch {
      alert('Oh No You Have No Success!');
    }
    history.push('/');
  };

  return (
    <div>
      <h1>Edit</h1>
      <DogForm
        name={name}
        setName={setName}
        bio={bio}
        setBio={setBio}
        image={image}
        setImage={setImage}
        age={age}
        setAge={setAge}
        breed={breed}
        setBreed={setBreed}
        handleSubmit={handleSubmit}
      />
    </div>
  );
}
