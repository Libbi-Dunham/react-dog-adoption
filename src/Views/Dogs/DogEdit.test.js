import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import DogEdit from './DogEdit';

test('renders dog information', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/2/edit']}>
      <Route path="/dogs/:id/edit" component={DogEdit} />
    </MemoryRouter>
  );
  await screen.findByDisplayValue('Luna');
  expect(container).toMatchSnapshot();
});
