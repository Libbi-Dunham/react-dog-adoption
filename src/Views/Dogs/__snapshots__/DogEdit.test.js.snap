// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dog information 1`] = `
<div>
  <div>
    <h1>
      Edit
    </h1>
    <div
      class="dog-form"
    >
      <form>
        <h2>
          Enter Dog Details
        </h2>
        <div
          class="form-control"
        >
          <label>
            Name: 
          </label>
          <input
            placeholder="Name"
            type="text"
            value="Luna"
          />
        </div>
        <div
          classsname="form-control"
        >
          <label>
            Bio: 
          </label>
          <textarea
            placeholder="bio"
          >
            High energy toy loving girl
          </textarea>
        </div>
        <div
          class="form-control"
        >
          <label>
            Age: 
          </label>
          <input
            placeholder="number"
            type="text"
            value="4"
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Image: 
          </label>
          <input
            placeholder="Image URL"
            type="text"
            value="https://placedog.net/samples/63.jpg"
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Breed: 
          </label>
          <input
            placeholder="breed"
            type="text"
            value="Golden retriever"
          />
        </div>
        <div
          class="action"
        >
          <button>
            Save
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
